### DISCUSSION TITLE
Pstgresql12 database connection pool issue

### DISCUSSION THREAD

**Discussion by kanchangpr** [prisma/prisma#4735] [Q&A] [answered]
I have created PostgreSQL 12 database cluster in digital ocean, and using connection pool. in my java code Application.properties i have configured database properties as below:

> `spring.datasource.url=jdbc:postgresql://private-db-postgresql--.b.db.ondigitalocean.com:25061/*********?pgbouncer=true
> 
> spring.datasource.username=*********
> 
> spring.datasource.password=*********
> 
> spring.datasource.driver-class-name=org.postgresql.Driver
> 
> spring.jpa.properties.hibernate.default_schema=*********`
> 

when i am trying to run the application getting below error:

> Caused by: org.postgresql.util.PSQLException: ERROR: prepared statement "S_3" already exists
> at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2553) ~[postgresql-42.2.16.jar:42.2.16]
> at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2285) ~[postgresql-42.2.16.jar:42.2.16]
> at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:323) ~[postgresql-42.2.16.jar:42.2.16]
> at org.postgresql.jdbc.PgConnection.executeTransactionCommand(PgConnection.java:857) ~[postgresql-42.2.16.jar:42.2.16]
> at org.postgresql.jdbc.PgConnection.rollback(PgConnection.java:900) ~[postgresql-42.2.16.jar:42.2.16]
> at com.zaxxer.hikari.pool.ProxyConnection.rollback(ProxyConnection.java:396) ~[HikariCP-3.4.5.jar:na]
> at com.zaxxer.hikari.pool.HikariProxyConnection.rollback(HikariProxyConnection.java) ~[HikariCP-3.4.5.jar:na]
> at org.hibernate.resource.jdbc.internal.AbstractLogicalConnectionImplementor.rollback(AbstractLogicalConnectionImplementor.java:121) ~[hibernate-core-5.4.21.Final.jar:5.4.21.Final]
> ... 27 common frames omitted
> Caused by: org.postgresql.util.PSQLException: ERROR: prepared statement "S_1" does not exist
> at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2553) ~[postgresql-42.2.16.jar:42.2.16]
> at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2285) ~[postgresql-42.2.16.jar:42.2.16]
> at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:323) ~[postgresql-42.2.16.jar:42.2.16]
> at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:473) ~[postgresql-42.2.16.jar:42.2.16]
> at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:393) ~[postgresql-42.2.16.jar:42.2.16]
> at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:164) ~[postgresql-42.2.16.jar:42.2.16]
> at org.postgresql.jdbc.PgPreparedStatement.executeQuery(PgPreparedStatement.java:114) ~[postgresql-42.2.16.jar:42.2.16]
> at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeQuery(ProxyPreparedStatement.java:52) ~[HikariCP-3.4.5.jar:na]

**Comment by janpio** [comment, marked as answer]
Is this question related to Prisma in any way?
If not, we will not be able to help you unfortunately.

**Reply by kanchangpr** [reply]
No, its not Prisma issue, its digitalOcean Database issue, raised issue in digitalOcean group.

**Reply by janpio** [reply]
Thanks and good luck.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
